apis/certificatemanager/v1beta1/certificatemanagerdnsauthorization_types.go

```diff
@@ -40,6 +40,11 @@ type CertificateManagerDNSAuthorizationSpec struct {
 	// +required
 	ProjectRef refs.ProjectRef `json:"projectRef"`
 
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Location field is immutable"
+	/* Immutable. Optional. Location represents the geographical location of the DnsAuthorization. If not specified, "global" is used. */
```

Comment:
" If not specified, "global" is used." -->  If so, maybe we should not use the immutable CRD validation, because using `spec.location: global` is actually the same as `spec.location` unspecified, but users cannot change it (which is confusing). I think we want to validate the location (and project) unchanged via the externalRef

Reply:
Or you can use kubebuilder defaulter. But I prefer handing this via the externalref.

Reply:
done. move the check to external ref